### PR TITLE
[CARE-1728][CARE-1858] Fix suggestions dropdown positioning

### PR DESCRIPTION
**Pull request by e1himself**

### CODE REVIEW

File: packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx [hunk 1]
```diff
@@ -37,36 +52,50 @@ export const MentionsDropdown = <V extends object>({
...
-
-            const rect = RangeFix.getBoundingClientRect(domRange);
-
-            if (!rect) {
-                return;
-            }
-
-            const { left, height, top } = rect;
-            suggestionsPanelRef.current.style.top = `${top + window.pageYOffset + height}px`;
-            suggestionsPanelRef.current.style.left = `${left + window.pageXOffset}px`;
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx:56]: This implementation based on `position: fixed` has stopped working after we've introduced a container element with `translate: 0 0` style (used by the new main navigation sidebar). This changes the containing block of the fixed-positioned element to the one with `translate: 0 0`, making the positioning not work anymore.

See https://developer.mozilla.org/en-US/docs/Web/CSS/position#fixed_positioning

File: packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx [hunk 2]
```diff
@@ -37,36 +52,50 @@ export const MentionsDropdown = <V extends object>({
...
+                if (!domRange) {
+                    return EMPTY_RECT;
+                }
+                const rect = RangeFix.getBoundingClientRect(domRange);
+                if (!rect) {
+                    return EMPTY_RECT;
+                }
+                const { left, right, top, bottom, width, height } = rect;
+                return {
+                    left,
```
Comment (MohammadxAli) [inline packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx:71]: Wouldn't a `...rest` spread be shorter and more suitable here?

**Review by MohammadxAli — APPROVED** [review]

**Review by kudlajz — APPROVED** [review]

File: packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx (hunk 2 above)
Comment (e1himself) [reply packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx]: Yeah, good point. This is not obvious, but we cannot use spread here. I've added a comment about it: 78268b68a09b5fd88a2ea6a9fc9be01f47b3ae38

Comment (MohammadxAli) [reply packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx]: Oh interesting, thanks!

**Pull request merged**